# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the Refresh / Content Opportunity Scoring lane.

I selected this lane because the starter dataset already contains visibility, traffic, freshness, and engagement signals. The goal is to identify which content pages should be reviewed first for refresh or optimization. This is useful because content teams have limited time and cannot manually review every page.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision:
Which content pages should be reviewed first?

Action:
A content or SEO reviewer examines the highest-ranked pages and decides whether to refresh, expand, improve CTR, or monitor them.

Unit of analysis:
One content page.

Output:
A ranked opportunity score and review queue.

Cost of a wrong recommendation:
Important pages may be ignored, leading to missed traffic opportunities. Time may also be wasted reviewing pages that do not need attention.

Why ML might help:
The dataset contains many signals such as impressions, clicks, sessions, CTR, content age, freshness, and engagement metrics. ML can combine these signals to prioritize pages more efficiently than manual review.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nAverage Impressions:")
print(df["impressions_90d"].mean())

print("\nAverage Sessions:")
print(df["sessions_90d"].mean())

Rows: 30000
Columns: 44

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average Impressions:
5200.3663

Average Sessions:
37.066633333333336


# Quick Look at the Data

The starter dataset contains 30,000 content pages and 44 features.

The average number of impressions over the last 90 days is approximately 5,200, indicating that many pages receive meaningful search visibility.

The average number of sessions is approximately 37 per page.

The most common trend category is **down** with 16,262 pages, suggesting that a large portion of the content inventory may be experiencing declining performance and could require review.

These observations suggest that prioritizing content review decisions is a meaningful problem and that data-driven methods may help identify which pages deserve attention first.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project may identify patterns associated with content decline or opportunity.

It cannot prove that refreshing a page will cause traffic growth.

The recommendations should be treated as decision-support signals rather than guaranteed outcomes.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.